src/pages/admin/admin.tsx

```diff
@@ -1,26 +1,63 @@
-import React, {FC} from 'react';
+import React, {FC, Fragment, useEffect, useState} from 'react';
 import { useForm } from 'react-hook-form';
+import {Cake} from '../../App';
 
 
-const Admin: FC = () => {
-	const { register, handleSubmit } = useForm(); // initialise the hook
-	const onSubmit = (data: any)=> {
+const AdminPage: FC = () => {
+	const { register, handleSubmit } = useForm();
+	const [cakes, setCakes] = useState<Cake[]>([]);
+	const [orders, setOrders] = useState([]);
+
+	useEffect (()=>{
+		setCakes(	JSON.parse(localStorage.getItem('cakes') as string));
+	},[]);
+
+	useEffect(()=>{
+		setOrders( JSON.parse(localStorage.getItem('orders') as string));
+	}, []);
+
```

Comment:
Since we're using context, these cakes and orders can be gotten from the context in app.tsx instead of having to get them i think here and possibly products page too. 

Also, even if using this method, the two things can be done in a single useEffect.